### PR TITLE
[Python] Remove duplication with hpp-fcl.

### PR DESCRIPTION
**Pull request by jmirabel**
Follows https://github.com/humanoid-path-planner/hpp-fcl/pull/133

Should help reducing compilation time too (e.g. https://github.com/stack-of-tasks/pinocchio/issues/1074)

### CODE REVIEW

**Comment by jmirabel** [comment]
CI complains about this
```
2020-02-21T15:40:06.7077036Z   File "/home/runner/work/pinocchio/pinocchio/examples/collisions.py", line 2, in <module>
2020-02-21T15:40:06.7077310Z     import pinocchio as pin, hppfcl
2020-02-21T15:40:06.7077905Z ImportError: No module named hppfcl
```
hppfcl is not available for CI ? If no, then this example should not be ran...

**Comment by gabrielebndn** [comment]
> CI complains about this
> 
> ```
> 2020-02-21T15:40:06.7077036Z   File "/home/runner/work/pinocchio/pinocchio/examples/collisions.py", line 2, in <module>
> 2020-02-21T15:40:06.7077310Z     import pinocchio as pin, hppfcl
> 2020-02-21T15:40:06.7077905Z ImportError: No module named hppfcl
> ```
> 
> hppfcl is not available for CI ? If no, then this example should not be run...

It seems the CI does not install the Python bindings:

https://github.com/stack-of-tasks/pinocchio/blob/66bb1bc08669497a0da3d303841e88f68b7434bb/travis_custom/custom_before_install#L26-L29

**Comment by jcarpent** [comment]
The Python bindings are located in robotpkg-pyxx-hpp-fcl

File: bindings/python/multibody/fcl/expose-fcl.cpp [hunk 1]
```diff
@@ -18,89 +13,8 @@ namespace pinocchio
...
-      typedef ::hpp::fcl::CachedMeshLoader CachedMeshLoader;
-
-      if(!eigenpy::register_symbolic_link_to_registered_type<MeshLoader>())
-        MeshLoaderPythonVisitor<MeshLoader>::
-        expose("Class to create CollisionGeometry from mesh files.");
-
-      if(!eigenpy::register_symbolic_link_to_registered_type<CachedMeshLoader>())
-        MeshLoaderPythonVisitor<CachedMeshLoader>::
-        expose("Class to create CollisionGeometry from mesh files with cache mechanism.");
-
```
Comment (jmirabel) [inline bindings/python/multibody/fcl/expose-fcl.cpp]: The implicit conversion, a few lines below, never gets called. This causes a unit test failure. @jcarpent, any idea why ?

      bp::implicitly_convertible< Transform3f,SE3 >();

Comment (jcarpent) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: @jmirabel You should keep the symbolink link to MeshLoader and so on.

Comment (jmirabel) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: I don't understand your answer. If I have to keep all the `if(!eigenpy::register_symbolic_link_to_registered_type) { ... }`, then this PR is useless...

Comment (jcarpent) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: This is why the tests are not passing because pinocchio.MeshLoader() does not exist. Because it is inside the scope hppfcl. I'm forcing that here for convenience.

Comment (jmirabel) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: I realized that and updated the examples and the unit test. The only thing which I do not know how to fix now is the conversion from SE3 to Transform3f.

Comment (jcarpent) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: I'm not able to see it in the test.
But again, I think important import hppfcl Meshloader, geometric entities inside Pinocchio scope. It will make life easy and will not breack the Python current API.

Comment (jmirabel) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: > I'm not able to see it in the test.

Ah, the failure on my computer and on CI is not the same... I just pushed a fix of the CI failure. On my computer, I still have a failure on bindings_fcl_transform.py

Comment (jcarpent) [reply bindings/python/multibody/fcl/expose-fcl.cpp]: @jmirabel Please keep shortcut to not break the current Python API.

**Comment by jmirabel** [comment]
Does anyone understand why on 16.04, travis finds robotpkg-hpp-fcl 1.1.3 and does not find robotpkg-py27-hpp-fcl. github-workflow finds the correct versions.

**Comment by jcarpent** [comment]
> Does anyone understand why on 16.04, travis finds robotpkg-hpp-fcl 1.1.3 and does not find robotpkg-py27-hpp-fcl ?

I think @nim65s may provide some guidance on it?

**Comment by nim65s** [comment]
@jmirabel : do you mean this https://travis-ci.org/stack-of-tasks/pinocchio/jobs/654927967#L1102 ?

In this case, this is 14.04, not 16.04: https://travis-ci.org/stack-of-tasks/pinocchio/jobs/654927967#L10

And it looks like Anthony finally removed 14.04 when he installed the new build server for robotpkg : http://robotpkg.openrobots.org/rbulk/robotpkg/index.html

Canonical dropped support for this almost a year ago,

**Comment by jmirabel** [comment]
> In this case, this is 14.04, not 16.04: https://travis-ci.org/stack-of-tasks/pinocchio/jobs/654927967#L10

Damned, I missed that... @jcarpent any other comments ?

**Comment by jmirabel** [comment]
If it is satisfactory, I propose to squash all the CI commits into one.

**Comment by jcarpent** [comment]
OK. It is fine for me. Please go ahead @jmirabel.

**Comment by jmirabel** [comment]
On my side, this is ready.
